### PR TITLE
Derive unnecessary imports analyzer from AbstractBuiltInCodeStyleDiagnosticAnalyzer

### LINKED ISSUE #62638: AbstractRemoveUnnecessaryImportsDiagnosticAnalyzer should sub-type AbstractBuiltInCodeStyleDiagnosticAnalyzer
https://github.com/dotnet/roslyn/blob/8bd10f1548db958b1983f3bd236fcc56aafae1be/src/Analyzers/Core/Analyzers/RemoveUnnecessaryImports/AbstractRemoveUnnecessaryImportsDiagnosticAnalyzer.cs#L21-L22

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes #62638

### CODE REVIEW

**Comment by Youssef1313** [comment]
@mavasani This is ready for review. Soon we'll get rid of all analyzers that derive directly from `DiagnosticAnalyzer`.

Remaining analyzers:

- `AbstractValidateFormatStringDiagnosticAnalyzer`: This will get deprecated in favor of CA2241 after work in roslyn-analyzers is done.
- `UnboundIdentifiersDiagnosticAnalyzerBase`: This will be removed with #62874
- `SimplifyTypeNamesDiagnosticAnalyzerBase`: This is being done in #62869.

File: src/Analyzers/Core/Analyzers/Helpers/DiagnosticHelper.cs [hunk 1]
```diff
@@ -265,6 +265,12 @@ static string EncodeIndices(IEnumerable<int> indices, int additionalLocationsLen
             if (id.StartsWith("JSON", StringComparison.Ordinal))
                 return null;
 
+            if (id == RemoveUnnecessaryImports.AbstractRemoveUnnecessaryImportsDiagnosticAnalyzer.DiagnosticFixableId)
+                return null;
+
+            if (id == "IDE0005_gen")
```
Comment (mavasani) [inline src/Analyzers/Core/Analyzers/Helpers/DiagnosticHelper.cs:271]: Add a comment for both these cases?

**Review by mavasani — APPROVED** [review]

File: src/Analyzers/Core/Analyzers/Helpers/DiagnosticHelper.cs (hunk 1 above)
Comment (Youssef1313) [reply src/Analyzers/Core/Analyzers/Helpers/DiagnosticHelper.cs:271]: I was just preserving the original behavior, and have no idea why/if this behavior is correct.

Can help link for `IDE0005_gen` be visible? The diagnostic is hidden and not configurable, so help link can never be shown?

Same applies to `AbstractRemoveUnnecessaryImportsDiagnosticAnalyzer.DiagnosticFixableId`.

Comment (mavasani) [reply src/Analyzers/Core/Analyzers/Helpers/DiagnosticHelper.cs:271]: In that case, can we return null for both cases?

Comment (mavasani) [reply src/Analyzers/Core/Analyzers/Helpers/DiagnosticHelper.cs:271]: Probably add a comment on the same lines as your previous comment: `This diagnostic is hidden and not configurable, so help link can never be shown and is not applicable`

**Review by mavasani — APPROVED** [review]

**Comment by Youssef1313** [comment]
@mavasani This should be ready to merge now.

**Pull request merged**